src/os/ogc/ogcservice.js

```diff
@@ -0,0 +1,397 @@
+goog.module('os.ogc.OGCService');
+goog.module.declareLegacyNamespace();
+
+const Promise = goog.require('goog.Promise');
+const Uri = goog.require('goog.Uri');
+const GeoJSON = goog.require('ol.format.GeoJSON');
+const Request = goog.require('os.net.Request');
+const OGCFilterModifier = goog.require('os.ogc.filter.OGCFilterModifier');
+const OGCFilterOverride = goog.require('os.ogc.filter.OGCFilterOverride');
+const WFSFormatter = goog.require('os.ogc.wfs.WFSFormatter');
+const osOlFeature = goog.require('os.ol.feature');
+const AbstractService = goog.require('os.ui.AbstractService');
+const GMLParser = goog.require('os.ui.file.gml.GMLParser');
+const geo2 = goog.require('os.geo2');
+const spatial = goog.require('os.ogc.spatial');
+const OGCQuery = goog.require('os.ogc.query.OGCQuery');
+const OSSettings = goog.require('os.config.Settings');
+
+
+/**
+ * Opensphere's Settings Manager
+ */
+const Settings = OSSettings.getInstance();
+
+/**
+ * The field to look for label, etc
+ * @type {string}
+ */
+const DEFAULT_ID_FIELD = 'NAME';
+
+/**
+ * Provides a service to get a data from a WFS layer.
+ * @template T
+ */
+class OGCService extends AbstractService {
+  /**
+   * Constructor.
+   */
+  constructor() {
+    super();
+
+    /**
+     * Promise used to request all countries.
+     * @type {goog.Promise}
```

Comment:
Types declared like this should use the locally defined variable name instead of the global type declaration.

At a glance, there are quite a few places where this is happening in this PR, so I won't point them all out, but please keep an eye out for them.

Reply:
I preferred the full-resolution pathname in the jsdocs/casts so that:
1) among multiple files we'll always have the same string to search for, and
2) there's no inconsistencies/swapping back and forth for "unused" requires (for JSDoc/Type-ing only) and ones where we actually "use" the Object/Utilities in the require
> e.g. 
> If I only had "osx.blah.Blah" in a Type-ing comment, the require at the top of the doc would be goog.require('osx.blah.Blah');
> const a = /** @type {osx.blah.Blah} */ ({...});
> 
> Buf If I actually create a Blah at some point, then I have to go through and replace a bunch...
> const Blah = goog.require('osx.blah.Blah');
> const a = /** @type {Blah} */ ({...});
> const b = new Blah();
Whatcha think?